covid/extract.py

```diff
@@ -44,3 +45,59 @@ def get_state_abbreviations_to_names():
         abbreviations = json.load(state_abbreviations_file)
 
     return abbreviations
+
+
+def power_bi_extractor(response):
+    data = json.loads(response.text)
+    value_list = data['results'][0]['result']['data']['dsr']['DS'][0]['PH'][1]['DM1']
+    for vl in value_list:
+        yield vl['C']
+
+
+# CDC data source: https://www.cdc.gov/nhsn/covid19/report-patient-impact.html
+
+def extract_cdc_inpatient_beds():
+    # State Representative Estimates for Percentage of Inpatient Beds Occupied (All Patients)
+    response = requests.post(
+        ugc.URL,
+        headers={**ugc.BASE_HEADERS, **ugc.INPATIENT_BED_HEADERS},
+        data=open("./covid/extract_config/inpatient_bed_query.json"))
+
+    df = pd.DataFrame(
+        power_bi_extractor(response),
+        columns=["State", "inpatient_bed_percent_occupied", "inpatient_beds_occupied"])
+
+    df = df.set_index('State')
+
+    return df
+
+
+def extract_cdc_icu_beds():
+    # State Representative Estimates for Percentage of ICU Beds Occupied (All Patients)
+    response = requests.post(
+        ugc.URL,
+        headers={**ugc.BASE_HEADERS, **ugc.ICU_BED_HEADERS},
+        data=open("./covid/extract_config/icu_bed_query.json"))
+
+    df = pd.DataFrame(
+        power_bi_extractor(response),
+        columns=["State", "icu_percent_occupied", "icu_beds_occupied"])
+
+    df = df.set_index('State')
+
+    return df
+
+
+def extract_cdc_facilities_reporting():
```

Comment:
@BriceWolfgang this is great data to pull. @lucasmbrown: this data is not in Category 3, but I think it is worth reporting alongside those numbers. 80% availability with only a few hospitals reporting is much different than 80% with 100% reporting, etc. What do you think? I can just pull the info regardless I guess.

Reply:
Realized we can also do that in a separate PR for the transform/loading so we could remove it from scope here.

Reply:
I agree with the sentiment though, would definitely be interesting to have those numbers as well!